packages/workspace/src/parser/parse.ts

```diff
@@ -48,3 +49,22 @@ export const parse = async (
     ? parseBufferAndFixErrors(srcString, filename, functions)
     : parseBuffer(srcString, filename, functions)
 }
+
+export type Token = {
+  value: string
+  type: string
+  col: number
+  line: number
+}
+
+export function *tokenizeContent(content: string): IterableIterator<Token> {
+  const lexer = new PeekableLexer(content)
+  try {
+    while (true) {
+      const token = lexer.next()
+      yield { value: token.value, type: token.type, col: token.col, line: token.line }
+    }
+  } catch (e) {
+    // Lexer throws an error when there is nothing left to parse
+  }
```

Comment:
seems like the correct usage for iterating the lexer is using the form
```typescript
while (lexer.peek()) {
 ...
}
```
not using exceptions for flow control

so this can be
```suggestion
  while (lexer.peek()) {
    yield lexer.next()
  }
```